### PR TITLE
doc(davide) add env tutorial for bsuite env

### PR DESCRIPTION
**Pull request by davide97l**
English version of the documentation of bsuite environment.

### CODE REVIEW

File: source/env_tutorial/bsuite.rst [hunk 1]
```diff
@@ -0,0 +1,273 @@
+Atari
+~~~~~~~
+
+概述
+=======
+
+Atari是最经典最常用的离散动作空间强化学习环境，常来作为离散空间强化学习算法的基准测试环境。它是一系列环境的集合（共有57个子环境），常用的子环境有Pong，Space
+Invaders，QBert，Enduro，Breakout，MontezumaRevenge等等，下图所示为其中的Space Invaders游戏。
```
Comment (norman26625) [inline source/env_tutorial/bsuite.rst]: Atari是最经典最常用的离散动作空间强化学习环境，常作为离散空间强化学习算法的基准测试环境。它是一个由Pong，Space，Invaders，QBert，Enduro，Breakout，MontezumaRevenge等57个子环境构成的集合，下图所示为其中的Space Invaders游戏。

File: source/env_tutorial/bsuite.rst [hunk 2]
```diff
@@ -1,273 +1,190 @@
-Atari
+Bsuite
 ~~~~~~~
 
-概述
+Description
 =======
 
-Atari是最经典最常用的离散动作空间强化学习环境，常来作为离散空间强化学习算法的基准测试环境。它是一系列环境的集合（共有57个子环境），常用的子环境有Pong，Space
-Invaders，QBert，Enduro，Breakout，MontezumaRevenge等等，下图所示为其中的Space Invaders游戏。
```
Comment (norman26625) [inline source/env_tutorial/bsuite.rst]: Atari是最经典最常用的离散动作空间强化学习环境，常作为离散空间强化学习算法的基准测试环境。它是一个由Pong，Space，Invaders，QBert，Enduro，Breakout，MontezumaRevenge等57个子环境构成的集合，下图所示为其中的Space Invaders游戏。

File: source/env_tutorial/bsuite.rst [hunk 3]
```diff
@@ -1,273 +1,190 @@
...
-
-   -  5：FIRE
-
-.. _奖励空间-1:
-奖励空间
+.. _Rewards space-1:
+Rewards space
 --------
 
--  游戏得分，根据具体游戏内容不同会有非常大的差异，一般是一个\ ``float``\ 数值，具体的数值可以参考最下方的基准算法性能部分。
```
Comment (norman26625) [inline source/env_tutorial/bsuite.rst]: 根据具体游戏内容不同，游戏得分会有非常大的差异，通常情况下是一个\ ``float``\ 数值，具体的数值可以参考最下方的基准算法性能部分。

File: source/env_tutorial/bsuite.rst (hunk 2 above)
Comment (norman26625) [reply source/env_tutorial/bsuite.rst]: 再多介绍一下Space Invaders，游戏模式是什么样的，游戏的类型，环境中有xxxxx元素，主要用来xxxx

File: source/env_tutorial/bsuite.rst [hunk 4]
```diff
@@ -1,273 +1,190 @@
...
 
--  游戏结束即为当前环境episode结束
+-  Environments terminate once they have reached their maximum number of steps or is encountered a failure state. All environments have a fixed number of maximum steps, but not all environments have a failure state.
 
-关键事实
+Key facts
 ========
 
-1. 2D
-   RGB三通道图像输入，但单帧图像蕴含的信息不足（比如运动方向），需要堆叠多帧图像来解决
```
Comment (norman26625) [inline source/env_tutorial/bsuite.rst]: 虽然是RGB三通道图像输入，但需要堆叠多帧图像来解决单帧图像蕴含的信息不足（例如运动方向）

File: source/env_tutorial/bsuite.rst [hunk 5]
```diff
@@ -1,273 +1,190 @@
...
-
-2. 离散动作空间
+1. Each environment contains several configurations to make it gradually more challenging.
 
-3. 既有稠密奖励（Space
-   Invaders）；又有稀疏奖励（Pitfall，MontezumaRevenge）
```
Comment (norman26625) [inline source/env_tutorial/bsuite.rst]: xxxx（需要主语）的奖励类型主要有两种：稠密奖励(Space Invaders)、稀疏奖励(Pitfall，MontezumaRevenge)

File: source/env_tutorial/bsuite.rst [hunk 6]
```diff
@@ -1,273 +1,190 @@
...
+2. Discrete actions space.
 
-4. 奖励取值尺度变化较大
```
Comment (norman26625) [inline source/env_tutorial/bsuite.rst]: 融合进 line97吧
  xxxx（需要主语）的奖励类型主要有两种：稠密奖励(Space Invaders)、稀疏奖励(Pitfall，MontezumaRevenge)，这两种奖励取值的尺度变化较大

File: source/env_tutorial/bsuite.rst [hunk 7]
```diff
@@ -1,273 +1,190 @@
...
-
--  变换内容：灰度图，空间尺寸缩放，最大最小值归一化，堆叠相邻N个游戏帧（N=4）
-
--  变换结果：三维np数组，尺寸为\ ``(4, 84, 84)``\ ，即为相邻的4帧灰度图，数据类型为\ ``np.float32``\ ，取值为 ``[0, 1]``
-
-.. _动作空间-2:
-动作空间
---------
-
--  基本无变换，依然是大小为N的离散动作空间，但一般为一维np数组，尺寸为\ ``(1, )``\ ，数据类型为\ ``np.int64``
```
Comment (norman26625) [inline source/env_tutorial/bsuite.rst]: 依然是大小为N的离散动作空间，但一般为一维np数组，尺寸为\ ``(1, )``\ ，数据类型为\ ``np.int64``

File: source/env_tutorial/bsuite.rst [hunk 8]
```diff
@@ -1,273 +1,190 @@
...
 .. code:: python
-   import gym
-   obs_space = gym.spaces.Box(low=0, high=1, shape=(4, 84, 84), dtype=np.float32)
-   act_space = gym.spaces.Discrete(6)
-   rew_space = gym.spaces.Box(low=-1, high=1, shape=(1, ), dtype=np.float32)
-.. _其他-2:
-其他
-----
-
--  ``epsiode_life``\ ：训练时的环境使用\ ``episode_life``\ 选项，即环境拥有多条生命值（一般为5），原始环境游戏失败一次生命值减一，所有生命值耗尽才视为episode结束
```
Comment (norman26625) [inline source/env_tutorial/bsuite.rst]: ``epsiode_life``\ ：训练时的环境使用\ ``episode_life``\ 选项，即环境拥有多条生命值（一般为5），在原始环境下游戏失败一次生命值减一，游戏生命值耗尽则视为episode结束

File: source/env_tutorial/bsuite.rst [hunk 9]
```diff
@@ -1,273 +1,190 @@
...
+   from bsuite import sweep
+   print('List bsuite_id for "bandit_noise" experiment:')
+   print(sweep.BANDIT_NOISE)
 
-训练和测试环境的区别
---------------------
+.. image:: ./images/bsuite_config.png
+   :align: center
 
--  训练环境使用动态随机种子，即每个episode的随机种子都不同，都是由一个随机数发生器产生，但这个随机数发生器的种子是通过环境的\ ``seed``\ 方法固定的；测试环境使用静态随机种子，即每个episode的随机种子相同，通过\ ``seed``\ 方法指定。
```
Comment (norman26625) [inline source/env_tutorial/bsuite.rst]: -  训练环境使用动态随机种子，即每个episode的随机种子都不同（由一个随机数发生器产生），但这个随机数发生器的种子是通过环境的\ ``seed``\ 方法固定的；测试环境使用静态随机种子，即每个episode的随机种子相同，通过\ ``seed``\ 方法指定。

**Comment by norman26625** [comment]
pls notice that when modifying the Chinese doc, synchronize and adjust the English doc

File: source/env_tutorial/bsuite.rst [hunk 10]
```diff
@@ -0,0 +1,190 @@
+Bsuite
+~~~~~~~
+
+Description
+=======
+
+``bsuite`` is a collection of carefully-designed experiments that investigate core capabilities of a reinforcement learning (RL) agent with two main objectives:
+
+    -  1. To collect clear, informative and scalable problems that capture key issues in the design of efficient and general learning algorithms.
+    -  2. To study agent behavior through their performance on these shared benchmarks.
```
Comment (Will-Nie) [inline source/env_tutorial/bsuite.rst:10]: choose either '-' representation or ordered list

File: source/env_tutorial/bsuite.rst [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+
+.. image:: ./images/bsuite.png
+   :align: center
```
Comment (Will-Nie) [inline source/env_tutorial/bsuite.rst:13]: I guess we need to add reference for these graphs

File: source/env_tutorial/bsuite.rst [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+   timestep = env.reset()
+   print(timestep)
+Original environment space
+========================
+
+.. _Observations space-1:
+Observations space
+--------
+
+-  Array representing the state of the environment, dimensions and size can vary according to the specific environment, datatype is \ ``np.float32``.
```
Comment (Will-Nie) [inline source/env_tutorial/bsuite.rst:48]: ..specific environment. Its datatype is \ ``np.float32``.

File: source/env_tutorial/bsuite.rst [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+   cfg = EasyDict(cfg)
+   memory_len_env = BSuiteEnv(cfg)
+
+
+DI-zoo runnable code
+====================
+
+The full training configuration can be found on `github
+link <https://github.com/opendilab/DI-engine/tree/main/dizoo/bsuite/config/serial>`__
+. Following, we show an example of configuration for the file, ``memory_len_0_dqn_config.py``\, you can run the demo with the following code：
```
Comment (Will-Nie) [inline source/env_tutorial/bsuite.rst:132]: In the following part, we show an example of configuration for the file, ``memory_len_0_dqn_config.py``\, you can run the demo with the following code：

File: source/env_tutorial/bsuite.rst [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+Key facts
+========
+
+1. Each environment contains several configurations to make it gradually more challenging.
+
+2. Discrete actions space.
+
+3. Each environment is designed to test a particular propriety of RL policies, including: generalization, exploration, credit assignment, scaling, noise, memory.
+
+4. Rewards scale can vary a lot.
```
Comment (Will-Nie) [inline source/env_tutorial/bsuite.rst:83]: Replace 'a lot' by 'significantly' since 'a lot' is a little bit informal and oral

Comment (Will-Nie) [reply source/env_tutorial/bsuite.rst:83]: The scale of rewards might be better

File: source/env_tutorial/bsuite.rst [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+
+-  Array representing the state of the environment, dimensions and size can vary according to the specific environment, datatype is \ ``np.float32``.
+
+.. _Actions space-1:
+Actions space
+--------
+
+-  The action space is a discrete space of size N, (where N can vary according to the environment), datatype is \ ``int``\ and input is a python integer value（or a np array of dimension 0 such as \ ``np.array(1)``\ to input action 1）.
+
+-  For example, in the Deep Sea environment, N is equal to 2, thus action values ranges from 0 to 1. For their specific meaning you can refer to the following list.
```
Comment (Will-Nie) [inline source/env_tutorial/bsuite.rst:56]: I was wondering why there was a 'thus' after 'N is equal to 2'. Is there a link?
For their specific meaning,  you can refer to the following list. (add a comma)

File: source/env_tutorial/bsuite.rst [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+
+   -  0：LEFT.
+
+   -  1：RIGHT.
+
+.. _Rewards space-1:
+Rewards space
+--------
+
+-  Rewards are assigned according to the rules of the environments, rewards are usually a \ ``float``\ value.
```
Comment (Will-Nie) [inline source/env_tutorial/bsuite.rst:66]: A full stop after 'the rules of the environments'

File: source/env_tutorial/bsuite.rst [hunk 17]
```diff
@@ -0,0 +1,190 @@
...
+--------
+
+Configurations are designed to increase the level of difficulty of an environment. For example, in a 5-armed bandit environment, configurations are used to regulate the level of noise to perturb the rewards.
+Given a specific environment, all possible configurations can be visualized with the following code snippet.
+
+.. code:: python
+
+   from bsuite import sweep
+   print('List bsuite_id for "bandit_noise" experiment:')
+   print(sweep.BANDIT_NOISE)
```
Comment (Will-Nie) [inline source/env_tutorial/bsuite.rst:111]: I believe the should be generated through the following code snippet:
from bsuite import sweep
for bsuite_id in sweep.BANDIT_NOISE:
  env = bsuite.load_from_id(bsuite_id)
  print('bsuite_id={}, settings={}, num_episodes={}'
        .format(bsuite_id, sweep.SETTINGS[bsuite_id], env.bsuite_num_episodes))

File: source/env_tutorial/bsuite.rst [hunk 18]
```diff
@@ -0,0 +1,190 @@
...
+
+About configurations
+--------
+
+Configurations are designed to increase the level of difficulty of an environment. For example, in a 5-armed bandit environment, configurations are used to regulate the level of noise to perturb the rewards.
+Given a specific environment, all possible configurations can be visualized with the following code snippet.
+
+.. code:: python
+
+   from bsuite import sweep
```
Comment (Will-Nie) [inline source/env_tutorial/bsuite.rst:109]: Maybe explain more about sweep? Something like 'This module ontains `bsuite_ids` for each experiment individually via uppercase constants corresponding to the experiment name'

File: source/env_tutorial/bsuite.rst [hunk 19]
```diff
@@ -0,0 +1,190 @@
...
+Rewards space
+--------
+
+-  Rewards are assigned according to the rules of the environments, rewards are usually a \ ``float``\ value.
+
+.. _Other-1:
+Others
+----
+
+-  Environments terminate once they have reached their maximum number of steps or is encountered a failure state. All environments have a fixed number of maximum steps, but not all environments have a failure state.
```
[the remark this answers is not in the document]
Comment (davide97l) [reply source/env_tutorial/bsuite.rst:72]: I would change 'the' with 'their', what do you think?

File: source/env_tutorial/bsuite.rst (hunk 15 above)
Comment (davide97l) [reply source/env_tutorial/bsuite.rst:56]: 'thus' is to explain that 'N is equal to 2' causes that 'action values ranges from 0 to 1'

Comment (davide97l) [reply source/env_tutorial/bsuite.rst:56]: why comma and not . or : ?

File: source/env_tutorial/bsuite.rst (hunk 17 above)
Comment (davide97l) [reply source/env_tutorial/bsuite.rst:111]: Yes, this code provides more information

File: source/env_tutorial/bsuite.rst (hunk 18 above)
Comment (davide97l) [reply source/env_tutorial/bsuite.rst:109]: added as comment: from bsuite import sweep  # this module contains information about all the environments

**Pull request merged**